content/blog/getting-started-with-k8s-part1/index.md

```diff
@@ -0,0 +1,261 @@
+---
+title: "Getting Started With Kubernetes: Clusters and Nodes"
+date: 2020-06-23
+meta_desc: "A tutorial for getting started with Kubernetes using Infrastructure as Code."
+meta_image: getting-started.png
+authors:
+    - sophia-parafina
+tags:
+    - Kubernetes
+
+---
+
+Containers solved the problem of moving software from one environment to another because they encapsulate all the software dependencies. However, an orchestration platform is needed to manage containers at scale. [Kubernetes](https://kubernetes.io/) is a popular open-source solution that uses declarative configuration to specify the desired state of the application. Configuring and deploying an application on Kubernetes is often accomplished with YAML files to define the state and command line tools to manage and control the Kubernetes API. This article demonstrates how to use infrastructure as code to create [basic Kubernetes objects](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#kubernetes-objects) and higher-level abstractions that build upon the basic objects.
+
+<!--more-->
+
+This is the first article of a multi-part series covering deploying Kubernetes and applications using infrastructure as code. We'll take a top-down approach starting from clusters to high-level abstractions such as deployments. This article aims to provide a step-by-step example of deploying an application in Kubernetes using infrastructure as code. For this example, we'll use Typescript for the programming language and provide sample code for AWS, Azure, GCP, and Kubernetes.
+
+## Clusters and Nodes
+
+A cluster is formed with a [control plane](https://kubernetes.io/docs/concepts/#kubernetes-control-plane) and a collection of [nodes](https://kubernetes.io/docs/concepts/architecture/nodes/), the smallest unit of computing in Kubernetes. A node can be either a physical or virtual that contains the necessary [components](https://kubernetes.io/docs/concepts/overview/components/#node-components) to run containers. A cluster has one or more nodes designated to the control plane, which controls the worker nodes where the application containers are deployed. A control plane node has:
+
+- the Kube-APIServer, which provides communication to the cluster
+- a Kube-Controller-Manager that provides governance for the cluster
+- etcd, which is the cluster state database
+- Kube-Scheduler, which schedules worker nodes based on events in etcd
+
+To create a cluster for Kubernetes on a cloud provider, we need to create a VPC that will host the nodes required to deploy the Kubernetes cluster. Each cloud provider requires configuration specific to their implementation, and we'll cover their particular requirements.
+
+{{< chooser cloud "aws,azure,gcp" >}}
```

Comment:
would we be able to add digitalocean here, perhaps? It's relatively easy to deploy a cluster to DO so might be worth adding?

Reply:
Not for this blog, but a DO post would be a good addition.